src/Build/Evaluation/Evaluator.cs

```diff
@@ -1383,6 +1418,15 @@ private void EvaluatePropertyElement(ProjectPropertyElement propertyElement)
         {
             using (_evaluationProfiler.TrackElement(propertyElement))
             {
+                if (propertyElement.Name.Equals(ReservedPropertyNames.allProjects, StringComparison.OrdinalIgnoreCase)
```

Comment:
This is not the hottest of paths, but still hot, so it might show up on perf. I'll run a perf test and see how it looks.

Worst case, we need to reuse the pattern from other string matchings in hot paths, like this one: [https://github.com/Microsoft/msbuild/blob/master/src/Shared/Modifiers.cs#L93-L111](https://github.com/Microsoft/msbuild/blob/master/src/Shared/Modifiers.cs#L93-L111)